crates/rpc/ipc/src/server/mod.rs

```diff
@@ -198,30 +206,58 @@ where
                     };
 
                     let service = self.http_middleware.service(tower_service);
-                    connections.add(Box::pin(spawn_connection(
+                    connections.push(tokio::spawn(spawn_connection(
                         ipc,
                         service,
                         stop_handle.clone(),
                         rx,
-                    )));
+                    ).in_current_span()));
 
                     id = id.wrapping_add(1);
+                    stopped = stop;
                 }
-                Err(MonitoredError::Selector(err)) => {
-                    tracing::error!("Error while awaiting a new IPC connection: {:?}", err);
+                AcceptConnection::Shutdown => { break; }
+                AcceptConnection::Err((e, stop)) => {
+                    tracing::error!("Error while awaiting a new IPC connection: {:?}", e);
+                    stopped = stop;
                 }
-                Err(MonitoredError::Shutdown) => break,
             }
         }
 
-        connections.await;
+        // FuturesUnordered won't poll anything until this line but because the
+        // tasks are spawned (so that they can progress independently)
+        // then this just makes sure that all tasks are completed before
+        // returning from this function.
+        while connections.next().await.is_some() {}
```

Comment:
this is unreachable and results in unbounded mem

please check jsonrpsee server loop on main

https://github.com/paritytech/jsonrpsee/blob/master/server/src/server.rs

Reply:
Yeah, but would like merge it first, and fire a new PR later today.

Reply:
okay sg, could you please open an issue for this then?

Reply:
https://github.com/paradigmxyz/reth/issues/7916 
would fire a PR today.